certora/specs/MintBurnEthereum.spec

```diff
@@ -0,0 +1,133 @@
+// This is spec is taken from the Open Zeppelin repositories at https://github.com/OpenZeppelin/openzeppelin-contracts/blob/448efeea6640bbbc09373f03fbc9c88e280147ba/certora/specs/ERC20.spec, and patched to support the DelegationToken.
+
+import "Delegation.spec";
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Rules: only the token holder or an approved third party can reduce an account's balance                             │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule onlyAuthorizedCanTransfer(env e, method f) {
```

Comment:
why is this rule in mint/burn

edit: probably because it makes sense in Optimism

Reply:
Yes, the TLDR here is that rule for Optimism is written differently because the `burn` function doesn't have the same restrictions in both implementations.
In the Ethereum impl, only `msg.sender` can burn his tokens. In the Optimism version the only the manager may "burn" on behalf of a user to send it back to the mainnet.